pkg/agent/controller/networkpolicy/reject.go

```diff
@@ -117,6 +118,23 @@ func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
 
 	sIface, srcFound := c.ifaceStore.GetInterfaceByIP(srcIP)
 	dIface, dstFound := c.ifaceStore.GetInterfaceByIP(dstIP)
+
+	isFlexibleIPAMSrc, isFlexibleIPAMDst, ctZone, err := parseFlexibleIPAMStatus(pktIn, c.nodeConfig, srcIP, srcFound, dstIP, dstFound)
+	if err != nil {
+		return err
+	}
+	matches := pktIn.GetMatches()
+	if (isFlexibleIPAMSrc || isFlexibleIPAMDst) && (srcFound && dstFound) {
+		// Set dstFound to false if the reject packet srcIP and dstIP is local but the original packet is coming from uplink.
+		// This case occurs for an ingress reject rule with AntreaIPAM cross-VLAN traffic, the reject type is RejectLocalToRemote.
+		if match := matches.GetMatchByName(binding.OxmFieldInPort); match != nil {
+			switch match.GetValue().(uint32) {
+			case c.nodeConfig.UplinkNetConfig.OFPort, c.nodeConfig.GatewayConfig.OFPort:
```

Comment:
The code comment says "uplink", but here both uplink and gateway OFPort are checked. Update the comments?

Reply:
Added `gw` to comment.